contracts/actions/curve/CurveWithdraw.sol

```diff
@@ -0,0 +1,141 @@
+        if (_nCoins == 4) return bytes4(0xa6929895);
+        if (_nCoins == 5) return bytes4(0xcbfe789f);
+        if (_nCoins == 6) return bytes4(0xfe27e085);
+        if (_nCoins == 7) return bytes4(0x78afd240);
+        if (_nCoins == 8) return bytes4(0xef930778);
+        revert("Invalid number of coins in pool.");
+    }
+
+    struct Params {
+        address sender;
+        address receiver;
+        address pool;
+        address lpToken;
+        uint256 burnAmount;
+        uint256[] minAmounts;
+        address[] tokens;
+        bool useUnderlying;
+    }
+
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+        params.sender = _parseParamAddr(params.sender, _paramMapping[0], _subData, _returnValues);
+        params.receiver = _parseParamAddr(params.receiver, _paramMapping[1], _subData, _returnValues);
+        params.pool = _parseParamAddr(params.pool, _paramMapping[2], _subData, _returnValues);
+        params.lpToken = _parseParamAddr(params.lpToken, _paramMapping[3], _subData, _returnValues);
+        params.burnAmount = _parseParamUint(params.burnAmount, _paramMapping[4], _subData, _returnValues);
+        
+        uint256 N_COINS = params.minAmounts.length;
+        require(N_COINS == params.tokens.length);
+        for (uint256 i = 0; i < params.tokens.length; i++) {
+            params.minAmounts[i] = _parseParamUint(params.minAmounts[i], _paramMapping[5 + i], _subData, _returnValues);
+            params.tokens[i] = _parseParamAddr(params.tokens[i], _paramMapping[5 + N_COINS + i], _subData, _returnValues);
+        }
+
+        _curveWithdraw(params);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public payable virtual override {
+        Params memory params = parseInputs(_callData);
+
+        _curveWithdraw(params);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Dont forget NatSpec
+    function _curveWithdraw(Params memory _params) internal {
+        _params.lpToken.pullTokensIfNeeded(_params.sender, _params.burnAmount);
+        _params.lpToken.approveToken(_params.pool, _params.burnAmount);
+        
+        uint256[] memory balances = new uint256[](_params.tokens.length);
+        for (uint256 i = 0; i < _params.tokens.length; i++) {
+            balances[i] = _params.tokens[i].getBalance(address(this));
+        }
+
+        bytes memory payload = _constructPayload(_params.minAmounts, _params.burnAmount, _params.useUnderlying);
+        (bool success, ) = _params.pool.call(payload);
```

Comment:
Is there a way to validate if curve pool?

Reply:
keep mapping of pool addresses in View contract and add it to refistr?